detectree2/models/train.py

```diff
@@ -433,74 +695,62 @@ def load_json_arr(json_path):
-        workers:
-        ims_per_batch:
-        gamma:
-        backbone_freeze:
-        warm_iter:
-        momentum:
-        batch_size_per_im:
-        base_lr:
-        max_iter:
-        num_classes:
-        eval_period:
-        out_dir:
-    """
-    cfg = get_cfg()
-    cfg.merge_from_file(model_zoo.get_config_file(base_model))
-    cfg.DATASETS.TRAIN = trains
-    cfg.DATASETS.TEST = tests
-    cfg.DATALOADER.NUM_WORKERS = workers
-    #cfg.SOLVER.IMS_PER_BATCH = ims_per_batch
-    cfg.SOLVER.GAMMA = gamma
-    cfg.MODEL.BACKBONE.FREEZE_AT = backbone_freeze
-    cfg.SOLVER.WARMUP_ITERS = warm_iter
-    cfg.SOLVER.MOMENTUM = momentum
-    #cfg.MODEL.RPN.BATCH_SIZE_PER_IMAGE = batch_size_per_im
-    #cfg.SOLVER.WEIGHT_DECAY = 0.001
-    cfg.SOLVER.BASE_LR = base_lr
-    cfg.OUTPUT_DIR = out_dir
-    os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)
-    if update_model is not None:
-        cfg.MODEL.WEIGHTS = update_model
-    else:
-        cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url(base_model)
-
-    cfg.SOLVER.IMS_PER_BATCH = ims_per_batch
-    cfg.SOLVER.BASE_LR = base_lr
-    cfg.SOLVER.MAX_ITER = max_iter
-    cfg.MODEL.ROI_HEADS.NUM_CLASSES = num_classes
-    cfg.TEST.EVAL_PERIOD = eval_period
-    return cfg
-
+def setup_cfg(
+    base_model="COCO-InstanceSegmentation/mask_rcnn_R_101_FPN_3x.yaml",
+    trains=("trees_train",),
+    tests=("trees_val",),
+    update_model=None,
+    workers=2,
+    ims_per_batch=1,
+    base_lr=0.0003,
+    max_iter=1000,
+    num_classes=1,
+    eval_period=100,
+    out_dir="/content/drive/Shareddrives/detectree2/train_outputs"):
+  """
+  To set up config object
+  """
+  cfg = get_cfg()
+  cfg.merge_from_file(model_zoo.get_config_file(base_model))
+  cfg.DATASETS.TRAIN = trains 
+  cfg.DATASETS.TEST = tests
+  cfg.DATALOADER.NUM_WORKERS = workers
+  cfg.OUTPUT_DIR = out_dir
+  cfg.SOLVER.IMS_PER_BATCH = 2
+  cfg.SOLVER.GAMMA = 0.1
+  cfg.MODEL.BACKBONE.FREEZE_AT = 3
+  cfg.SOLVER.WARMUP_ITERS = 120
+  cfg.SOLVER.MOMENTUM = 0.9
+  cfg.MODEL.RPN.BATCH_SIZE_PER_IMAGE = 128
+  cfg.SOLVER.WEIGHT_DECAY = 0
+  cfg.SOLVER.BASE_LR = 0.001
+  cfg.betas = (0.9, 0.999)
+  os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)
+  if update_model is not None:
+    cfg.MODEL.WEIGHTS = update_model # DOESN'T WORK
+  else:
+    cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url(base_model)
+  
+  cfg.SOLVER.IMS_PER_BATCH = ims_per_batch
+  cfg.SOLVER.BASE_LR = base_lr
+  cfg.SOLVER.MAX_ITER = max_iter
+  cfg.MODEL.ROI_HEADS.NUM_CLASSES = num_classes
+  cfg.TEST.EVAL_PERIOD = eval_period
+  cfg.MODEL.BACKBONE.FREEZE_AT = 2
+  #cfg.MODEL.ROI_BOX_HEAD.BBOX_REG_LOSS_TYPE = 'diou'
+  cfg.MODEL.PROPOSAL_GENERATOR.NAME  = 'custom_RPN'
+  cfg.nms_thresh_union = 0.2
+  cfg.MODEL.RPN.NMS_THRESH = 0
+
+  #cfg.nms_thresh_union = 0
+  #cfg.MODEL.ROI_HEADS.NMS_THRESH_TEST = 0.1
+
+  #cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.5
+  cfg.MODEL.ROI_HEADS.NUM_CLASSES = 1  # only has one class (pnumonia)
+  return cfg
 
```

Comment:
the change in predictions_on_data is simply to change the predictor from DelaultTrainer to the new predictor in custom_nms.py